compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -380,6 +385,14 @@ public void visit(BLangFunction funcNode) {
         } finally {
             this.finalizeCurrentWorkerActionSystem();
         }
+        funcNode.annAttachments.forEach(annotationAttachment -> analyzeNode(annotationAttachment, env));
+    }
+
+    private void validateParams(BLangFunction funcNode) {
+        funcNode.requiredParams.forEach(param -> analyzeNode(param, env));
```

Comment:
Shall we use a plain old foreach instead?

Reply:
Done